test/functional/api/vim_spec.lua

```diff
@@ -115,6 +115,15 @@ describe('API', function()
       nvim('exec','autocmd BufAdd * :let x1 = "Hello"', false)
       nvim('command', 'new foo')
       eq('Hello', request('nvim_eval', 'g:x1'))
+
+      -- Script scope (s:)
+      eq('ahoy! script-scoped varrrrr', nvim('exec', [[
+          let s:pirate = 'script-scoped varrrrr'
+          func! s:avast_ye_hades(s) abort
+            return s.' '.s:pirate
+          endf
+          echo <sid>avast_ye_hades('ahoy!')
+        ]], true))
```

Comment:
maybe we also want a test for the scope remaining after the fact. I mean a global function returning another `s:` var. and a second `eq` using `nvim_eval` or something (which should not have the `s:` scope set, could even test that `s:pirate` _fails_ in that context).

Reply:
I have added 2 more tests. Do they take care of the cases?